core/services/mavftp/main.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+
+import errno
+import os
+import time
+from argparse import ArgumentDefaultsHelpFormatter, ArgumentParser
+from typing import Any, Dict, Optional
+
+from fuse import FUSE, FuseOSError, LoggingMixIn, Operations
+from loguru import logger
+from pymavlink import mavutil
+
+from mavftp import FTPModule
+
+# this allows us to support acessing the special files
+base_file_paths = ["/", "@ROMFS", "@SYS"]
+
+
+class MAVFTP(LoggingMixIn, Operations):
+    def __init__(self, mav: Any) -> None:
+        self.mav = mav
+        self.files: Dict[str, Dict[str, int]] = {
+            path: {
+                "st_mode": 0o40755,
+                "st_nlink": 2,
+                "st_size": 0,
+                # we need some timestamp for filebrowser be happy
+                "st_ctime": int(time.time()),
+                "st_mtime": int(time.time()),
+                "st_atime": int(time.time()),
+            }
+            for path in base_file_paths
+        }
+        self.ftp = FTPModule(mav)
+
+    def fix_path(self, path: str) -> str:
+        if path.startswith("/@"):
+            return path[1:]
+        return path
+
+    def getattr(self, path: str, _fh: int = 0) -> Any:
```

Comment:
can we have a better name than _fh ?

Reply:
these are mostly the exact definition from the examples in fusepy, but I agree they are not descriptive, I'll change to file_handler